pvlib/pvsystem.py

```diff
@@ -1133,6 +1121,28 @@ def calcparams_desoto(effective_irradiance, temp_cell,
          Source: [4]
     '''
 
+    # test for use of function pre-v0.6.0 API change
+    if isinstance(a_ref, dict) or \
+       (isinstance(a_ref, pd.Series) and ('a_ref' in a_ref.keys())):
+        import warnings
+        warnings.warn('module_parameters detected as fourth positional'
+                      + ' argument of calcparams_desoto')
+        warnings.warn('calcparams_desoto will require one argument for' \
+                      + ' each module model parameter in v0.7.0 and later')
+        try:
+            module_parameters = a_ref
+            a_ref = module_parameters['a_ref']
+            I_L_ref = module_parameters['I_L_ref']
+            I_o_ref = module_parameters['I_o_ref']
+            R_sh_ref = module_parameters['R_sh_ref']
+            R_s = module_parameters['R_s']
+            warnings.warn('module parameters extracted from fourth positional'\
+                          + ' argument')
+        except:
```

Comment:
Recommend something like

```
except Exception as e:
    raise e('Module parameters could not be extracted from '
            '4th positional argument. Check that parameters are '
            'from CEC database and/or update your code for new API.')
```

and then no need for the warning or return below

Reply:
OK